### PR TITLE
Nerfs and Reworks Ninja

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Justification

To me it doesn't really make sense that a mid-round ghost role can have so much power over affecting the round while being nearly untouchable. 
- Most players are uneasy about hunting Ninjas because it has a chance to be NanoTrasen aligned. While it really isn't against the rules to kill a NanoTrasen ninja because it's a ninja, people are still hesitant to do so anyways like they're hesitant to toolbox a Delta Station engineer who stumbles upon the station.
- Ninjas are super fast and incredibly stealthy. It is absurdly hard to catch one and is very close to wizard in terms of evasion. This is fine for a ninja, but it is really strong for a mid-round ghost role.
- Not only do Ninjas (potentially) have objectives to kill or protect heads, but they (potentially) have objectives to kill or protect antagonists, effectively outing them or shutting down their round. No mid-round ghost role should have a license to validhunt.
- Ninjas have a license to capture people and effectively keep them from the round entirely unless they die. This is pretty anti-fun and is honestly no different than round-removing random people.

# Document the changes in your pull request

- Ninjas no longer have an alliance (NanoTrasen aligned, Syndicate Aligned) and will always be Neutral.
- Ninjas can have up to 4 objectives chosen from a pool of objectives (See below).
- Ninja suit stats are no longer randomized.
- Energy nets no longer teleport players to GBJ (shoutout to SimpleFlips!), but rather trap them for 30 seconds.
- Stealth mode require less than 0.5 lumens (0 is completely dark, 1 is completely fullbright) to work.
- Increases the battery cell of the suit from a cell that gives 9000 energy to a cell that gives 20000 energy.
- Increases the energy cost of EMP from 2500 to 5000.
- Increases the energy cost of Adrenaline from 0 to 250.
- Decreases the energy cost of Nets from 2500 to 1000.
- Increases the energy cost of Smoke Bombs from 0 to 100.
- Increases the energy cost of Shuriken from 300 to 500.
- Increases the energy cost of Sword Recall from 100 per turf to 200 per turf.
- Increases the suit passive energy cost from 5 per tick to 10 per tick.
- Increases the suit stealth energy cost from 25 per tick to 50 per tick.
- Increases the amount of adrenal boosters from 3 to 5.

Current Ninja objectives (4 randomly chosen, can have duplicates).
- Steal research nodes.
- Steal important item.
- Kill (anyone, previously it was limited to heads/antags). Has a 20% chance to additionally add an "extract with a photo of the body" objective.
- Protect (anyone, previously it was limited to heads/antags). Has a 20% chance to additionally add a "prevent the target from escaping" objective.

With a guaranteed 

- Survive until the end of the round.

# Wiki Documentation

https://wiki.yogstation.net/wiki/Space_Ninja Needs to be updated.

# Changelog

:cl:  BurgerBB
tweak: Rebalances Ninja and their objectives.
/:cl:

### CODE REVIEW

**Comment by BurgerLUA** [comment]
Forgot to convert to draft because none of this is tested (or even compiled)

**Comment by Moja10** [comment]
Lol it doesn't make any sense for a ninja to go on the shuttle. It was perfectly fine with them staying on the station, they are not affiliated for or against Nanotrasen, so why would they go to CC? 

GBJ has always been a key feature of ninjas, I would suggest just adding a method for them to return, ala contractor. 

Alliances are a neat addition, I wish alliances were added to more antags, not less. If anything, the core objectives should just change to better distinguish. Maybe more protects/captures for NT Ninja, more kill, steal for Syndie Ninjas. 

Also die glorious doesn't make sense for ninja, the lad has literally a priceless billion dollar suit and sword and the higher ups want him to kill himself? lol

**Comment by RG4ORDR** [comment]
Gonna address some of your points.
>Players valid hunt the ninja
Yeah this happens because AFAIK ninjas have no rules protecting them, they're still an antag.
>Ninja are very fast and stealthy 
Yes because ninjas are a glass cannon and require constant upkeep of their resources. They need to plan ahead for everything such as keeping their suit powered so they can maintain their high speed,stealth, throwing stars,nets,shockgloves, their teleports are tied to their katana. Not to mention if they get hit/get hungry they HAVE to turn their suit off unless they got piercing syringes, and hyposprays. Wizards aren't remotely as restricted as this.
>Ninja objectives are stupid.
Yeah true lol.
>Ninjas round remove people.
Controversial but true. Capturing people isn't as easy as players can break the net with relatively no effort, ninja has to be right next to the player to make sure the capture will go through. Granted when the Ninja dies everyone gets returned to the station, though adding a timer like contractors would be pretty good.

Otherwise all your changes are just not good and are conducive of someone whose never played ninja.
Ninja doesn't need die a glorious death otherwise there wouldn't be many ninjas to hire.

**Comment by BurgerLUA** [comment]
The only thing I agree with changing is the shuttle thing because I misread the code but other than that I disagree with the feedback.

**Comment by RG4ORDR** [comment]
> The only thing I agree with changing is the shuttle thing because I misread the code but other than that I disagree with the feedback.

I highly encourage you to play ninja.
Lemme give you a few counters to ninjas in no specific order.

- Temp Guns - hit them with freeze and they lose all that speed.
- Syringe guns - Not anymore after some bad PR
- Lasers - Ninja armor isn't strong
- Bullets - A shotgun can GG a ninja no problem, anything else can handle it very easily.
- Circular saw - Ninja armor isn't that strong you can cause bleed wounds VERY easily on ninjas.
- Knives - Ninja armor isn't that strong.
- Null rod and Nullrod variants - Ninja armor isn't that strong
- Riot suit and shield 
- Batons of any variety beat ninja.
- Ion gun - Bye bye power
- Emp grenades - Bye bye power.
- Throwing stars - Oh you thought if you missed you're fine? Boom, brute damage + bleeding + embed chance.
- Anything that can embed actually
- Xray or thermal eyes - Kinda hard to heal and eat when you're being hunted constantly.

Ninja is HIGHLY dependent on an individual players own skill and game knowledge. All it's abilities have some trade off, cool down, and you need to manage several other functions. Your adrenaline burst irradiates you and gives you toxins, and can knock you out. Your suit cannot beat on when you heal or eat except if you get hyposprays and medipens. I've seen good and bad players take ninja and the difference between a Ninja actually cause massive problems and dying within 5 minutes is the response of the security team and the valid hunters/people defending themselves from the ninja.

**Comment by Sniblet** [comment]
Asking a ninja to escape on the shuttle, especially when stealth mode doesn't work in shuttle lighting, effectively means they have hijack 50% of the time and die glorious the other 50%

**Comment by BurgerLUA** [comment]
The survive objective was re-added and the shuttle/die objective was removed.

Mid round antagonist ghost roles are supposed to be hard. Playing Blob is supposed to be hard. Playing a sentient virus is supposed to be hard. Playing vine spawns are supposed to be hard. Playing ninja is supposed to be hard, but honestly it feels like it belongs in the "wizard" category in terms of "hahaha you can't touch me, I remove you from the round!" when it comes to their insane evasion, their ability to send people to GBJ, and the fact that they usually have objectives that tell them to steal the brains of active players or to decap them for no raisin and space their head.

File: code/modules/antagonists/ninja/ninja.dm
Comment (ReddicusDragon) [inline code/modules/antagonists/ninja/ninja.dm:10]: restore this line

File: code/modules/antagonists/ninja/ninja.dm
Comment (ReddicusDragon) [inline code/modules/antagonists/ninja/ninja.dm:40]: why are you removing station ninja???

File: code/modules/antagonists/ninja/ninja.dm
Comment (ReddicusDragon) [inline code/modules/antagonists/ninja/ninja.dm:131]: once again... why?

**Review by ReddicusDragon — CHANGES_REQUESTED** [review]
I've read enough, what is this pr why are we removing station ninja

File: code/modules/antagonists/ninja/ninja.dm
Comment (ReddicusDragon) [reply code/modules/antagonists/ninja/ninja.dm:10]: restoring this line should probably solve lints

**Comment by BurgerLUA** [comment]
> I've read enough, what is this pr why are we removing station ninja

Station ninja was basically a license to validhunt antags which really isn't fair to the antags.

**Comment by ReddicusDragon** [comment]
> > I've read enough, what is this pr why are we removing station ninja
> 
> Station ninja was basically a license to validhunt antags which really isn't fair to the antags.

I'm requesting a review from Jamie, as I am not experienced nor knowledgeable to properly review this.

**Comment by ReddicusDragon** [comment]
seems like the kind of thing a headcoder should weigh in on

**Comment by BurgerLUA** [comment]
> seems like the kind of thing a headcoder should weigh in on

Hence the draft.

Needs a lot of input on what to change and what to keep.

**Comment by UltraBigRat** [comment]
honestly looks pretty good as the power cap increase balances out. Since you're reworking it could maybe give em a nutriment pump while yer at it? eating is very painful as ninja as you have to go through the long reboot sequence, which usually just leads to ninjas teleporting into space to do it. Or just let em eat through their mask and make helmet toggleable?

**Comment by Moja10** [comment]
still think this is a wonky PR, remember, this isn't competitive legends league, it's okay to have some midrounds be stronger than others, just like in chess. its not a game of "oh sentient virus is weak therefore ninja must be weak too"

**Comment by ReddicusDragon** [comment]
this is still a draft. Don't know why the ART maintainer is closing a CODE PR

**Comment by solwashere** [comment]
Don't like ninja objectives being reworked to have standard traitor objectives
I think reworking some of the already existing objectives would be much better, and making it so that Ninjail is more like contractor jail instead of straight up round removal until the ninja maybe dies 
I do agree that the NT ninja can suck for antags if they become a target, but I do think there could be some sort of workaround for that, for example, the ninja can only have antags that have hit targets on head or sec as targets, and said traitor's  get some sort of heads up message and maybe extra TC to compensate and prepare upon ninja spawn
Regarding the heads as target objective for the syndie ninja, as someone who mains a command role it is mildly unfair at times, especially when there lack of sec or competent sec, but I don't think it's a big enough frustration to warrant removing it considering that you can counter ninja with temp gun and iron + uranium 
As for the claim that people are too scared to validhunt the ninja, this is almost entirely false in all of the times that I've seen a ninja as a lot of people tend to be scared of the risk of the ninja round removing them regardless of which side they are on
tldr: I don't think we should whitewash the ninja and make it just traitor with cool sword + armor and we should have more unique objectives for antags

File: code/modules/antagonists/ninja/ninja.dm
Comment (BurgerLUA) [reply code/modules/antagonists/ninja/ninja.dm:10]: I did the opposite and just removed it as it is obsolete.

File: code/modules/antagonists/ninja/ninja.dm
Comment (BurgerLUA) [reply code/modules/antagonists/ninja/ninja.dm:40]: Station ninja was just validhunting antags and thus bad.

**Comment by BurgerLUA** [comment]
I might get more forum feedback or something. I don't know what I can do to make ninja not shit that people won't hate.

I'd prefer a headcoder to weigh in on this so I can gauge what to actually do.

**Comment by MajManatee** [comment]
Open for a month, no pushes for 1 week. Please re-open when work is continuing.

**Pull request closed without merging**
